frontend/src/components/Announcement/Announcement.jsx

```diff
@@ -14,24 +15,25 @@ import {
 
 const Announcement = ({ announcement, isOpen, showBoard }) => {
   const { title, iconType, boardName, content, comments, id } = announcement;
+  const { register } = useForm();
 
   const commentsText =
     comments && (comments.length === 1 ? '[1 comment]' : `[${comments.length} comments]`);
 
   return (
     <Wrapper>
-      <AnnouncementIcon iconType={iconType} />
+      <AnnouncementIcon iconType={iconType} {...register('icon', {required: true})}/>
       <TextWrapper>
-        <Board>{showBoard && boardName}</Board>
-        <Title>{title}</Title>
+        <Board {...register('boardName', {required: true})}>{showBoard && boardName}</Board>
+        <Title {...register('title', {required: true})}>{title}</Title>
         {isOpen ? (
-          <Content>{content}</Content>
+          <Content {...register('content', {required: true})}>{content}</Content>
         ) : (
           <>
-            <ShortContent>{content}</ShortContent>
-            <StyledLink to={`/announcement/${id}`}>
-              <ReadMore>[read more]</ReadMore>
-              <Comments>{commentsText}</Comments>
+            <ShortContent {...register('shortContent', {required: true})}>{content}</ShortContent>
+            <StyledLink {...register('link', {required: true})} to={`/announcement/${id}`}>
+              <ReadMore {...register('boardName')}>[read more]</ReadMore>
+              <Comments {...register('commentText')}>{commentsText}</Comments>
             </StyledLink>
           </>
         )}
```

Comment:
To jest komponent announcementu ale nie formularza. Tutaj się nic nie wysyła. Walidację trzeba dodać w miejscu, gdzie tworzymy nowy announcement i wysyłamy POST na serwer, a nie gdzie wyświetlamy już utworzony.

Reply:
Potwierdzam